### ISSUE TITLE
SFTP Issues After Upgrading To 0.5.4 (Daemon)

### ISSUE THREAD

**Issue by gamemann** [pterodactyl/panel#1035]
* Panel or Daemon: Daemon
* Version of Panel/Daemon: 0.5.4 (latest)
* Server's OS: Ubuntu 16.04.4 LTS
* Your Computer's OS & Browser: Windows 10 64-bit & Google Chrome

------------------------

Hey everyone, I have a couple issues. Yesterday I upgraded the control panel from 0.6.x to 0.7.5 (I do not remember what daemon version I upgraded from but I am running on 0.5.4 now). I currently have a RCON tool running on the same machine as the containers and it basically continuously retrieves a log file from one of our servers through SFTP using the credentials the control panel provides us with. Before the update, things ran fine but after the update, the RCON tool doesn't retrieve the log file correctly (the SFTP credentials were updated and I confirmed that it connects and is pointed towards the correct file path). It retrieves a part of the log file when the service is booted up but stops retrieving anything afterwards. My question is, were there any adjustments to the SFTP from those versions listed above? Perhaps a timeout that could have been added. The RCON tool I am running is called B3 RCON and it's for the game Call of Duty 4.

I would also like to note that SFTP is slower after the update as well. We also ran into an error that caused one of our nodes to go down. I had to restart the Wings service (the Wings service itself was still running on the machine) for the node to come back up on the control panel. This is the error I believe that caused the node to go down:

```
{"name":"wings","hostname":"game","pid":2188,"level":60,"err":{"message":"Cannot read property 'on' of undefined","name":"TypeError","stack":"TypeError: Cannot read property 'on' of undefined\n    at FileSystem.size (/srv/daemon/src/controllers/fs.js:83:20)\n    at Timeout.diskUse [as _onTimeout] (/srv/daemon/src/controllers/server.js:569:17)\n    at ontimeout (timers.js:390:15)\n    at tryOnTimeout (timers.js:250:5)\n    at Timer.listOnTimeout (timers.js:214:5)"},"msg":"A fatal error occured during an operation.","time":"2018-03-07T13:32:52.376Z","v":0}
{"name":"wings","hostname":"game","pid":2188,"level":60,"err":{"message":"spawn du EMFILE","name":"Error","stack":"Error: spawn du EMFILE\n    at exports._errnoException (util.js:1020:11)\n    at Process.ChildProcess._handle.onexit (internal/child_process.js:197:32)\n    at onErrorNT (internal/child_process.js:376:16)\n    at _combinedTickCallback (internal/process/next_tick.js:80:11)\n    at process._tickDomainCallback (internal/process/next_tick.js:128:9)","code":"EMFILE"},"msg":"A fatal error occured during an operation.","time":"2018-03-07T13:32:52.376Z","v":0}
```

I wasn't able to replicate this error afterwards.

If you need any other information, please feel free to ask!

Any help is highly appreciated!

Thanks!

**Comment by parkervcp** [comment]
So you are saying this only happened one time?

**Comment by DaneEveritt** [comment]
Couple of things here:

1.) "slower" without metrics means nothing, and isn't anything I can work off of;
2.) `EMFILE` indicates that your system is trying to open too many files at once, try looking into whats going on there to see what is happening;
3.) The process you're describing does not sound efficient at all, and I'm not sure I understand the reasoning behind what you're trying to do.

**Comment by gamemann** [comment]
Hello, I apologize for the late reply.

@parkervcp Yes, this only happened once. It hasn't happened since but I just wanted to report the error in case this was an issue with the control panel's code.

@DaneEveritt 
1) I agree. I just added it as a note in case it was reported before. I can try to get metrics in the future and probably open a separate issue. I apologize for that.

2) Thank you for that information. If it happens again, I will look into what's opening so many files at once.

3) The B3 RCON tool basically would retrieve a log file from one of my servers using SFTP. I would agree the process isn't efficient but at the time, I didn't know of a better way.

I discovered server data is stored in `/srv/daemon-data/`. I changed the B3 RCON tool to use the log file directly from `/srv/daemon-data/<containerID>/...`.

I apologize for the lack of data. I was just curious if there were any big changes made to the SFTP system in the control panel.

My main issue is resolved and I'm happy to be using a more efficient way to retrieve the log file from the server.

I will close this issue.

Thank you for your assistance!
